includes/wc-template-functions.php

```diff
@@ -948,6 +968,10 @@ function woocommerce_show_product_sale_flash() {
 	 */
 	function woocommerce_template_single_add_to_cart() {
 		global $product;
```

Comment:
The global $product actually gets used later in the function on these:

```do_action( 'woocommerce_' . $product->get_type() . '_add_to_cart' );```

so I thought it made sense to keep it as it was.